documents/bootstrapper.go

```diff
@@ -78,28 +77,35 @@ func (PostBootstrapper) Bootstrap(ctx map[string]interface{}) error {
 		return errors.New("document repository not initialised")
 	}
 
-	didService, ok := ctx[identity.BootstrappedDIDService].(identity.ServiceDID)
+	anchorRepo, ok := ctx[anchors.BootstrappedAnchorRepo].(anchors.AnchorRepository)
 	if !ok {
-		return errors.New("identity service not initialized")
+		return errors.New("anchor repository not initialised")
 	}
 
-	anchorRepo, ok := ctx[anchors.BootstrappedAnchorRepo].(anchors.AnchorRepository)
+	cfg, ok := ctx[bootstrap.BootstrappedConfig].(Config)
 	if !ok {
-		return errors.New("anchor repository not initialised")
+		return errors.New("documents config not initialised")
 	}
 
 	p2pClient, ok := ctx[bootstrap.BootstrappedPeer].(Client)
 	if !ok {
 		return errors.New("p2p client not initialised")
 	}
 
+	didService, ok := ctx[identity.BootstrappedDIDService].(identity.ServiceDID)
+	if !ok {
+		return errors.New("identity service not initialized")
+	}
+
+	ctx[BootstrappedAnchorProcessor] = DefaultProcessor(didService, p2pClient, anchorRepo, cfg)
+
 	txMan := ctx[transactions.BootstrappedService].(transactions.Manager)
 	anchorTask := &documentAnchorTask{
 		BaseTask: txv1.BaseTask{
 			TxManager: txMan,
 		},
 		config:        cfgService,
-		processor:     DefaultProcessor(didService, p2pClient, anchorRepo, cfg),
+		processor:     ctx[BootstrappedAnchorProcessor].(AnchorProcessor),
```

Comment:
lets create a variable for `DefaultProcessor` and use it here instead of type asserting here from ctx

Reply:
makes sense

Reply:
Sure. Will update it. Thanks for pointing it out :)